src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunk.cs

```diff
@@ -833,13 +861,15 @@ private enum CacheStatus {
 			if (!_inMem)
 				CreateReaderStreams();
 
-			_chunkFooter = ReadFooter(_writerWorkItem.WorkingStream);
 			IsReadOnly = true;
 
 			_writerWorkItem?.Dispose();
 			_writerWorkItem = null;
 
 			SetAttributes(_filename, true);
+
+			using var stream = _handle.AsUnbufferedStream(FileAccess.Read);
```

Comment:
is this supposed to work for InMem chunks? i _think_ so, but then we can't use _handle
is there a reason we can't use the _writerWorkItem now?

Reply:
> is this supposed to work for InMem chunks? i think so, but then we can't use _handle

nice catch! will fix it 

> is there a reason we can't use the _writerWorkItem now?

yes, because `ChunkDataWriteStream` cannot be used for reading (it throws `InvalidOperationException`)

Reply:
fixed

Reply:
ah got it. thanks